src/components/application_manager/src/resumption/resume_ctrl_impl.cc

```diff
@@ -914,6 +914,10 @@ void ResumeCtrlImpl::AddWayPointsSubscription(
         saved_app[strings::subscribed_for_way_points];
     if (true == subscribed_for_way_points_so.asBool()) {
       application_manager_.SubscribeAppForWayPoints(application);
+      auto message = MessageHelper::CreateMessageWithFunctionID(
+          application_manager_,
+          hmi_apis::FunctionID::Navigation_SubscribeWayPoints);
+      application_manager_.GetRPCService().ManageHMICommand(message);
```

Comment:
```suggestion
      if (!application_manager_.IsAnyAppSubscribedForWayPoints())
          auto message = MessageHelper::CreateMessageWithFunctionID(
              application_manager_,
              hmi_apis::FunctionID::Navigation_SubscribeWayPoints);
          application_manager_.GetRPCService().ManageHMICommand(message);
      }
      application_manager_.SubscribeAppForWayPoints(application);
```
You should check if any apps are already subscribed before sending this message